### PR TITLE
Core, Flink, Spark: Drop deprecated APIs scheduled for removal in 1.8.0

### PR DESCRIPTION
**Pull request by findepi**
Remove APIs that are deprecated and scheduled for removal in 1.8.0. Except for `SparkAppenderFactor` which is still used in `TestSparkMergingMetrics`. This should be addressed separately (https://github.com/apache/iceberg/pull/11076#pullrequestreview-2283613928).

### CODE REVIEW

**Review by jbonofre — APPROVED** [review]

**Comment by ajantha-bhat** [comment]
> Remove APIs that are deprecated and scheduled for removal in 1.8.0. Except for SparkAppenderFactor which is still used in TestSparkMergingMetrics. This should be addressed separately (https://github.com/apache/iceberg/pull/11076#pullrequestreview-2283613928).

`SparkAppenderFactory` can be removed as it is not used in production code. Thats what we concluded during deprecation. I will have a PR for that today.

**Comment by ajantha-bhat** [comment]
I created https://github.com/apache/iceberg/pull/11727 for `SparkAppenderFactor` now

**Review by ajantha-bhat — APPROVED** [review]
LGTM. 

Could have had the module wise PR for ease review.

File: flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/data/TestRowProjection.java [hunk 1]
```diff
@@ -369,19 +346,7 @@ public void testMapProjection() throws IOException {
     assertThat(projected.getMap(0)).isEqualTo(properties);
   }
 
-  private Map<String, ?> toStringMap(Map<?, ?> map) {
```
Comment (nastra) [inline flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/data/TestRowProjection.java:372]: removing this seems to be missing from the same file in Flink 1.19

Comment (findepi) [reply flink/v1.20/flink/src/test/java/org/apache/iceberg/flink/data/TestRowProjection.java:372]: the ide didn't flag it as unused there... turns out it was too lazy.

thanks for catching!

**Review by Fokko — APPROVED** [review]

**Review by nastra — APPROVED** [review]

**Pull request merged**